pkg/server/server.go

```diff
@@ -293,6 +293,54 @@ func mapSourceSpecToReponse(src client.Object) *pb.Source {
 	return source
 }
 
+func (s *applicationServer) AddApplication(ctx context.Context, msg *pb.AddApplicationRequest) (*pb.AddApplicationResponse, error) {
+	appUrl, err := gitproviders.NewNormalizedRepoURL(msg.Url)
+	if err != nil {
+		return nil, grpcStatus.Error(codes.InvalidArgument, fmt.Sprintf("unable to parse app url %q: %s", msg.Url, err))
```

Comment:
I suspect you could use https://pkg.go.dev/google.golang.org/grpc/status#Errorf here?

There are other cases where you're using `fmt.Sprintf` when you could use `Errorf`

Reply:
Nice catch. Fixed to use `Errorf` in that file.